hphp/runtime/test/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+enable_testing()
+find_package(GTest REQUIRED)
+include_directories(${GTEST_INCLUDE_DIRS})
+
+set(CXX_SOURCES)
+auto_sources(files "*.cpp" "RECURSE" "${CMAKE_CURRENT_SOURCE_DIR}")
+list(APPEND CXX_SOURCES ${files} )
+
+set(HEADER_SOURCES)
+auto_sources(files "*.h" "RECURSE" "${CMAKE_CURRENT_SOURCE_DIR}")
+list(APPEND HEADER_SOURCES ${files} "${CMAKE_CURRENT_SOURCE_DIR}/../../hhvm/process-init.cpp")
+
+add_executable(hphp_runtime_test ${CXX_SOURCES} ${HEADER_SOURCES})
+link_object_libraries(hphp_runtime_test ${HHVM_WHOLE_ARCHIVE_LIBRARIES})
+target_link_libraries(hphp_runtime_test ${HHVM_LINK_LIBRARIES} gtest gtest_main gmock gmock_main)
```

Comment:
So... GTest is a bit of an absurd beast... All you want here is `gtest_main`, everything else is defining the exact same symbols...